File: app/src/main/java/com/example/notificationapp/view/fragments/HomeFragment.java
package com.example.notificationapp.view.fragments;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.notificationapp.R;
import com.example.notificationapp.data.adapters.ClubListAdapter;
import com.example.notificationapp.data.network.api.RetrofitAccessObject;
import com.example.notificationapp.data.network.model.ClubModel;
import com.example.notificationapp.utils.Constants;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HomeFragment extends Fragment {
    private List<ClubModel> clubs = new ArrayList<>();
    private RecyclerView clubRecyclerView;
    private ClubListAdapter clubListAdapter;

    @SuppressLint("MissingInflatedId")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_home, container, false);
        clubRecyclerView = root.findViewById(R.id.clubRecycler);
        clubRecyclerView.setLayoutManager(new LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false));
        getClubsList();
        return root;
    }

    private void getClubsList() {
        SharedPreferences preferences = requireContext().getSharedPreferences(Constants.SHARED_PREFERENCE, Context.MODE_PRIVATE);
        RetrofitAccessObject.getRetrofitAccessObject()
                .getClubs(preferences.getString(Constants.TOKEN, ""))
                .enqueue(new Callback<List<ClubModel>>() {
                    @Override
                    public void onResponse(Call<List<ClubModel>> call, Response<List<ClubModel>> response) {
                        if (response.isSuccessful() && response.body() != null) {
                            clubs = response.body();
                            clubListAdapter = new ClubListAdapter(clubs, requireContext());
                            clubRecyclerView.setAdapter(clubListAdapter);
                            Log.d("Hello1",clubs.toArray().toString());
                        }
                    }
                    @Override
                    public void onFailure(Call<List<ClubModel>> call, Throwable t) {

                    }
                });
    }


}

File: app/src/main/java/com/example/notificationapp/view/fragments/HomeFragment.kt
package com.example.notificationapp.view.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import com.example.notificationapp.R
import com.example.notificationapp.app.UserInstance
import com.example.notificationapp.adapters.ClubListAdapter
import com.example.notificationapp.api.API
import com.example.notificationapp.api.ClubModel
import com.example.notificationapp.databinding.FragmentHomeBinding
import com.example.notificationapp.view.activities.MainActivity
import com.squareup.picasso.Callback
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import java.lang.Exception

class HomeFragment(private val mainActivity: MainActivity) : Fragment() {

    private var clubs: List<ClubModel> = ArrayList()
    private lateinit var clubListAdapter: ClubListAdapter

    private lateinit var _binding: FragmentHomeBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        val root = _binding.root
        _binding.clubRecycler.layoutManager = GridLayoutManager(requireContext(), 2)
        _binding.profileBtn.setOnClickListener { mainActivity.goToProfilePage() }

        getClubsList()
        setValues()
        return root
    }

    private fun getClubsList() {
        API.getClubs(UserInstance.getAuthToken(requireContext()), {
            clubs = it
            clubListAdapter = ClubListAdapter(clubs, requireContext())
            _binding.clubRecycler.adapter = clubListAdapter
        }) { Toast.makeText(requireContext(), "$it: Could not get clubs", Toast.LENGTH_SHORT).show() }
    }

    private fun setValues() {
        val avatar = UserInstance.getAvatar()
        if (avatar.isNotEmpty()) {
            Picasso.get().load(avatar).networkPolicy(NetworkPolicy.OFFLINE).placeholder(R.drawable.profile_icon)
                .into(_binding.profileIcon, object : Callback {
                    override fun onSuccess() {}
                    override fun onError(e: Exception?) {
                        Picasso.get().load(avatar).placeholder(R.drawable.profile_icon).into(_binding.profileIcon)
                    }
                })
        }
    }
}